### PR TITLE
Introduce DataSourceException

### PR DESCRIPTION
**Pull request by KochTobi**
Many thanks for contributing to this project!

**PR Checklist**
Please fill in the appropriate checklist below (delete whatever is not relevant). These are the most common things requested on pull requests (PRs).

- [x] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.rst` is updated

**Description of changes**
This PR introduces a DataSourceException that shall be thrown by a data source in case of technical malfunction.

### CODE REVIEW

**Comment by KochTobi** [comment]
I would argue that this does not need a changelog entry

File: sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/project/load/LoadProjectsDataSource.groovy [hunk 1]
```diff
@@ -16,6 +17,7 @@ interface LoadProjectsDataSource {
      * Loads projects from a user, given a user identifier
      * @return a list of projects the user has access to
      * @since 1.0.0
+     * @throws DataSourceException in case of a technical error with the data source
      */
-    List<Project> fetchUserProjects()
+    List<Project> fetchUserProjects() throws DataSourceException
```
Comment (jenniferboedker) [inline sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/project/load/LoadProjectsDataSource.groovy:22]: 👍🏼

File: sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/DataSourceException.groovy [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+     */
+    DataSourceException(String message) {
+        super(message)
+    }
+
+    /**
+     * Constructs a new runtime exception with the specified detail message and cause.
+     Note that the detail message associated with cause is not automatically incorporated in this runtime exception's detail message.
+
+     * @param message the detail message (which is saved for later retrieval by the getMessage() method).
```
Comment (jenniferboedker) [inline sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/DataSourceException.groovy:27]:
```suggestion
     * @param message the detailed message (which is saved for later retrieval by the getMessage() method).
```

File: sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/DataSourceException.groovy [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+     * @param cause  the cause (which is saved for later retrieval by the getCause() method).
+     *  (A null value is permitted, and indicates that the cause is nonexistent or unknown.)
```
Comment (jenniferboedker) [inline sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/DataSourceException.groovy:29]: if you add comments in '()' I would argue its not important enough. I would opt for either explicitly write it or not. (Just my personal preference)

```suggestion
     *  A null value is permitted, and indicates that the cause is nonexistent or unknown.
```

Comment (KochTobi) [reply sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/DataSourceException.groovy:29]: I have to be honest. The text was directly copied from RuntimeException :smile: I would leave it as is if this is fine with you?

**Review by jenniferboedker — APPROVED** [review]

**Pull request merged**